internal/provider/networks_appliance_vlans_resource.go

```diff
@@ -0,0 +1,654 @@
+			},
+		},
+	}
+}
+
+func (r *NetworksApplianceVlansResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*openApiClient.APIClient)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *NetworksApplianceVlansResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *NetworksApplianceVlansResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	object56 := openApiClient.NewInlineObject56(data.Id.ValueString(), data.Name.ValueString())
+	object56.SetCidr(data.Cidr.ValueString())
+	object56.SetId(data.VlanId.ValueString())
+	object56.SetApplianceIp(data.ApplianceIp.ValueString())
+	object56.SetGroupPolicyId(data.GroupPolicyId.ValueString())
+	object56.SetMask(int32(data.Mask.ValueInt64()))
+	object56.SetName(data.Name.ValueString())
+	object56.SetSubnet(data.Subnet.ValueString())
+	object56.SetTemplateVlanType(data.TemplateVlanType.ValueString())
+	ipv6 := openApiClient.NewNetworksNetworkIdApplianceSingleLanIpv6()
+	ipv6.SetEnabled(data.Ipv6.Enabled.ValueBool())
+	var prefixAssignments []openApiClient.NetworksNetworkIdApplianceSingleLanIpv6PrefixAssignments
+	for _, prefixAssignment := range data.Ipv6.PrefixAssignments {
+		originInterfaces := []string{}
+		for _, originInterface := range prefixAssignment.Origin.Interfaces.Elements() {
+			originInterfaces = append(originInterfaces, originInterface.String())
+		}
+		prefixAssignments = append(prefixAssignments, openApiClient.NetworksNetworkIdApplianceSingleLanIpv6PrefixAssignments{
+			Autonomous:         prefixAssignment.Autonomous.ValueBoolPointer(),
+			StaticPrefix:       prefixAssignment.StaticPrefix.ValueStringPointer(),
+			StaticApplianceIp6: prefixAssignment.StaticApplianceIp6.ValueStringPointer(),
+			Origin: &openApiClient.NetworksNetworkIdApplianceSingleLanIpv6Origin{
+				Type:       prefixAssignment.Origin.Type.ValueString(),
+				Interfaces: originInterfaces,
+			},
+		})
+	}
+	ipv6.SetPrefixAssignments(prefixAssignments)
+	object56.SetIpv6(*ipv6)
+
+	_, httpResp, err := r.client.ApplianceApi.CreateNetworkApplianceVlan(ctx, data.NetworkId.ValueString()).CreateNetworkApplianceVlan(*object56).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to update resource",
+			fmt.Sprintf("%v\n", err.Error()),
+		)
+	}
+
+	// collect diagnostics
+	if httpResp != nil {
+		tools.CollectHttpDiagnostics(ctx, &resp.Diagnostics, httpResp)
+	}
+
+	// Check for API success response code
+	if httpResp.StatusCode != 201 {
```

Comment:
update it to 200

Reply:
in the dashboard docs it says it will return 201 and not 200!

Reply:
in the dashboard page mentioned as 201 as creation response